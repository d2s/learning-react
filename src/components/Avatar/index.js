import React from 'react'
import avatarPicture from '../../images/d2s.jpg'

export default (props) => (
  <header className='tc pv4 pv5-ns'>
    <img src={avatarPicture} className='br-100 pa1 ba b--black-10 h3 w3' alt='avatar' />
    <h1 className='f5 f4-ns fw6 mid-gray'>{props.name}</h1>
    <h2 className='f6 gray fw2 ttu tracked'>{props.country}</h2>
  </header>
)