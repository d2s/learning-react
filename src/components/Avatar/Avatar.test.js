import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

import Avatar from './index';

describe('With ReactDOM', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Avatar />, div);
  });
})

describe('With Enzyme', () => {
  it('shallow renders without crashing', () => {
    shallow(<Avatar />);
  });
})

describe('With Snapshot Testing', () => {
  it('Avatar renders the same way', () => {
    const component = renderer.create(<Avatar />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})