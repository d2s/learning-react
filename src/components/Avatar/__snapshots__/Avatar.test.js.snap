exports[`With Snapshot Testing Avatar renders the same way 1`] = `
<header
  className="tc pv4 pv5-ns">
  <img
    alt="avatar"
    className="br-100 pa1 ba b--black-10 h3 w3"
    src="d2s.jpg" />
  <h1
    className="f5 f4-ns fw6 mid-gray avatar-name" />
  <h2
    className="f6 gray fw2 ttu tracked avatar-country" />
</header>
`;
